### ISSUE TITLE
Invalid Date object will throw a TypeError

### ISSUE THREAD

**Issue by dotnetCarpenter** [chaijs/loupe#58]
The example below will throw a ` TypeError` on [line 5 in date.js](https://github.com/chaijs/loupe/blob/master/lib/date.js#L5), because there is no guard against `dateObject.toJSON()` returning `null`. Which is the case for invalid dates.

### Reproduction

```js
const { inspect } = require ('loupe')

inspect (new Date ('not a date')) // <- TypeError: Cannot read properties of null (reading 'split')
```

Ref: https://github.com/vitest-dev/vitest/issues/2216

**Comment by dotnetCarpenter** [comment]
I would be happy to provide a patch if I know what the return value should be for invalid dates.
I am not a direct consumer of Loupe but rather of [Vitest](https://github.com/vitest-dev/vitest/). So for me it would make sense to return `NaN`, which is a global variable but that goes against the point of Loupe.

>Loupe turns the object you give it into a string.

A better place to fix this might be in [chai](https://github.com/chaijs/chai)?

**Comment by dotnetCarpenter** [comment]
Ping @keithamus

**Comment by dotnetCarpenter** [comment]
I think that for the sake of loupe, the right step forward is to return `Invalid Date`.

The specs unfortunately states that the behavior is both _"may fall back to any implementation-specific heuristics or implementation-specific date formats"_ **and** _"Strings that are unrecognizable shall cause Date.parse to return NaN"_.
This makes me think that we should return `NaN` since the String is _"unrecognizable"_ but my tests shows that both V8 and SpiderMonkey agrees on _"implementation-specific heuristics"_ for a String that is clearly not a date. It gets mucky when I use a String that has a year component in it. E.g. `foo-bar 2022`.

Below are my notes about the subject. I will add a simple PR which will return `"Invalid Date"` with a link to this issue.

From the [ECMAScript® 2022 Language Specification 21.4.2.1 Date ( ...values )](https://262.ecma-international.org/#sec-date)

When the Date function is called, the following steps are taken:

1. If NewTarget is undefined, then
    1. ...
2. Let numberOfArgs be the number of elements in values.
3. If numberOfArgs = 0, then
    1. ...
4. Else if numberOfArgs = 1, then
    1. Let value be values[0].
    2. If [Type](https://262.ecma-international.org/#sec-ecmascript-data-types-and-values)(value) is Object and value has a [[DateValue]] internal slot, then
        1. ...
    3. Else,
        1. Let v be ? [ToPrimitive](https://262.ecma-international.org/#sec-toprimitive)(value).
        2. If [Type](https://262.ecma-international.org/#sec-ecmascript-data-types-and-values)(v) is String, then
            1.[Assert](https://262.ecma-international.org/#assert): The next step never returns an [abrupt completion](https://262.ecma-international.org/#sec-completion-record-specification-type) because [Type](https://262.ecma-international.org/#sec-ecmascript-data-types-and-values)(v) is String.
            1. **Let tv be the result of parsing v as a date, in exactly the same manner as for the parse method ([21.4.3.2](https://262.ecma-international.org/#sec-date.parse)).**

[21.4.3.2](https://262.ecma-international.org/#sec-date.parse):
If the String does not conform to that format (_Date Time String Format_) the function may fall back to any implementation-specific heuristics or implementation-specific date formats. **Strings that are unrecognizable or contain out-of-bounds format element values shall cause `Date.parse` to return `NaN`.**

[21.4.4.37 Date.prototype.toJSON ( key )](https://262.ecma-international.org/#sec-date.prototype.tojson):

> Note 2
>
>The `toJSON` function is intentionally generic; it does not require that its this value be a Date. Therefore, it can be transferred to other kinds of objects for use as a method. However, it does require that any such object have a `toISOString` method.


### JS engine behavior

```js
const d = new Date('not a date');
```

| Method      | node 18 v8                 | Firefox                    |
| ----------- |:--------------------------:| :-------------------------:|
| valueOf     | `NaN`                      | `NaN`                      |
| toString    | Invalid Date               | Invalid Date               |
| toUTCString | Invalid Date               | Invalid Date               |
| toISOString | `RangeError: invalid date` | `RangeError: invalid date` | 
| toJSON      | `null`                     | `null`                     |


```js
const d = new Date('foo-bar 2022');
```

| Method      | node 18 v8                                                         | Firefox      |
| ----------- |:------------------------------------------------------------------:| :-----------:|
| valueOf     | `1640991600000`                                                    | `NaN`        |
| toString    | Sat Jan 01 2022 00:00:00 GMT+0100 (Central European Standard Time) | Invalid Date |
| toUTCString | Fri, 31 Dec 2021 23:00:00 GMT                                      | Invalid Date |
| toISOString | 2021-12-31T23:00:00.000Z                                           | Invalid Date | 
| toJSON      | 2021-12-31T23:00:00.000Z                                           | `null`       |


**Test code:**

```js
const test = f => {
	let r;
	try {
		r = f.call(d);
	} catch (err) {
		r = err;
	}
	return r;
}

[
    Date.prototype.valueOf,
	Date.prototype.toString,
	Date.prototype.toUTCString,
	Date.prototype.toISOString,
	Date.prototype.toJSON,
].map(test)
 .forEach(m => { console.debug(m); });
```

**Comment by dotnetCarpenter** [comment]
Fixed in [v2.3.5](https://github.com/chaijs/loupe/releases/tag/v2.3.5) 🎉
